src/components/inputs/z-otp/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, h, Event, EventEmitter, Prop } from "@stencil/core";
+
+@Component({
+  tag: "z-otp",
+  styleUrl: "styles.css",
+  shadow: true
+})
+export class ZOtp {
+  @Prop() inputNum?: number = 6;
+
+  private otp: string[];
+  private otpRef: HTMLInputElement[];
+
+  @Event() otpChange: EventEmitter;
+  emitInputChange(value: string) {
+    this.otpChange.emit({ value });
+  }
+
+  componentWillLoad() {
+    this.otp = Array.apply(null, Array(this.inputNum));
+    this.otpRef = Array.apply(null, Array(this.inputNum));
+  }
+
+  componentDidLoad() {
+    this.otpRef[0].focus();
+  }
+
+  render() {
+    return (
+      <div class="otp-container">
+        {this.otp.map((_elem, i) => {
+          return (
+            <input
+              onInput={(e: any) => {
+                this.otp[i] = e.target.value;
+                this.emitInputChange(this.otp.join(""));
+              }}
+              onKeyUp={(e: any) => {
+                if (e.keyCode !== 8 && e.keyCode !== 46)
+                  i < this.inputNum && this.otpRef[i + 1].focus();
+              }}
+              type="tel"
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel
`tel` should be a phone number, maybe `number` would be better?

Reply:
input number has a really ugly standard layout :sweat_smile: 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number

Reply:
I know, but you can hide the arrows :sweat_smile: